ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsNoJobTask.java

```diff
@@ -181,6 +186,15 @@ public void run() {
         }
         parameters.putAll(basicStatistics);
         StatsSetupConst.setBasicStatsState(parameters, StatsSetupConst.TRUE);
+        try {
+          String who = (partish.getPartition() == null) ? ("table " + partish.getTable().getFullyQualifiedName())
+                  : ("partition " + partish.getPartition().getName());
+          long threshold = conf.getLongVar(HiveConf.ConfVars.HIVE_MERGE_MAP_FILES_AVG_SIZE);
+          SmallFilesWarningUtil.smallFilesWarnings(parameters, 100L, threshold, who, "[ANALYZE][NOSCAN]")
+                                .ifPresent(smallFilesConsoleAndWarn(LOG));
+        } catch (Exception e) {
```

Comment:
if there is no exception assumed from the block, why not move the `try{ } catch{}`?